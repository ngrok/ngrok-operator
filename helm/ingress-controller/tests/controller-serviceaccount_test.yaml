suite: test controller-serviceaccount
templates:
- controller-serviceaccount.yaml
release:
  name: test-release
  namespace: test-namespace
tests:
- it: should create a serviceaccount
  asserts:
  - isKind:
      of: ServiceAccount
  - isAPIVersion:
      of: v1
  - hasDocuments:
      count: 1
- it: does not generate a serviceaccount when serviceAccount.create is false
  set:
    serviceAccount.create: false
  asserts:
  - hasDocuments:
      count: 0
- it: Should match the snapshot
  asserts:
  - matchSnapshot: {}
