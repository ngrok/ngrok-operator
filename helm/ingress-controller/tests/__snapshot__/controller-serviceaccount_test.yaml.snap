Should match the snapshot:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/component: controller
        app.kubernetes.io/instance: test-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: ngrok-ingress-controller
        app.kubernetes.io/part-of: ngrok-ingress-controller
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: ngrok-ingress-controller-0.2.0
      name: test-release-ngrok-ingress-controller
      namespace: test-namespace
